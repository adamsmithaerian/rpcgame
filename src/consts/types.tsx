export enum Shape {
  Rock,
  Paper,
  Scissors
}

export enum Result {
  Win,
  Lose,
  Tie
}

export const shapeNames: Array<String> = ['Rock', 'Paper', 'Scissors'];

export const resultNames: Array<String> = ['Win', 'Lose', 'Tie'];
