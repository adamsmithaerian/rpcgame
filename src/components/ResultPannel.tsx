import * as React from 'react';

import { Result, Shape, resultNames, shapeNames } from '../consts/types';

interface ResultPannelProps {
  result: Result;
  computerShape: Shape;
  userShape: Shape;
  onResetGame: () => void;
}

class ResultPannel extends React.Component<ResultPannelProps, any> {
  render() {
    return (
      <div className="Result-pannel">
        <div className="text">You chose {shapeNames[this.props.userShape]}</div>
        <div className="text">Comptuer chose {shapeNames[this.props.computerShape]}</div>
        <div className="text">You {resultNames[this.props.result]}</div>
        <button type="button" onClick={this.props.onResetGame}>Play Again</button>
      </div>
    );
  }
}

export default ResultPannel;
