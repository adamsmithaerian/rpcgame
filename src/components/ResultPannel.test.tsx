import * as React from 'react';
import * as ReactDOM from 'react-dom';
import * as renderer from 'react-test-renderer';

import ResultPannel from './ResultPannel';
import { Shape, Result } from '../consts/types';

describe('ResultPannel component should', () => {
  it('render without crashing', () => {
    const div = document.createElement('div');
  
    ReactDOM.render(
      <ResultPannel
        result={Result.Win}
        computerShape={Shape.Rock}
        userShape={Shape.Paper}
        onResetGame={() => {}}
      />,
      div
    );
  });

  it('match snapshot', () => {
    const tree = renderer
      .create(
        <ResultPannel
          result={Result.Win}
          computerShape={Shape.Rock}
          userShape={Shape.Paper}
          onResetGame={() => {}}
      />
    )
    .toJSON();
    expect(tree).toMatchSnapshot();
  });
});