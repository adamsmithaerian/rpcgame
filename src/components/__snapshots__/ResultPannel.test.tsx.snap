// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultPannel component should match snapshot 1`] = `
<div
  className="Result-pannel"
>
  <div
    className="text"
  >
    You chose 
    Paper
  </div>
  <div
    className="text"
  >
    Comptuer chose 
    Rock
  </div>
  <div
    className="text"
  >
    You 
    Win
  </div>
  <button
    onClick={[Function]}
    type="button"
  >
    Play Again
  </button>
</div>
`;
